src/components/datepicker/js/calendar.js

```diff
@@ -205,18 +200,42 @@
       angular.element(document.body).off('keydown', boundKeyHandler);
     });
 
-    if (this.minDate && this.minDate > $mdDateLocale.firstRenderableDate) {
+    // For Angular 1.4 and older, where there are no lifecycle hooks but bindings are pre-assigned,
+    // manually call the $onInit hook.
+    if (angular.version.major === 1 && angular.version.minor <= 4) {
+      this.$onInit();
+    }
+
+  }
+
+  /**
+   * Angular Lifecycle hook for newer Angular versions.
+   * Bindings are no longer assigned in the controller, but in the $onInit hook.
```

Comment:
Maybe: "Bindings are not guaranteed to have been assigned in the controller, but they are in the `$onInit` hook."